src/snowflake/snowpark/_internal/compiler/utils.py

```diff
@@ -0,0 +1,38 @@
+#!/usr/bin/env python3
+#
+# Copyright (c) 2012-2024 Snowflake Computing Inc. All rights reserved.
+#
+
+from typing import Dict, List
+
+from snowflake.snowpark._internal.analyzer.expression import Attribute
+from snowflake.snowpark._internal.analyzer.snowflake_plan import SnowflakePlan
+from snowflake.snowpark._internal.analyzer.snowflake_plan_node import (
+    SnowflakeCreateTable,
+)
+from snowflake.snowpark._internal.compiler.query_generator import QueryGenerator
+
+
+def create_query_generator(plan: SnowflakePlan) -> QueryGenerator:
```

Comment:
Is this util for test only? Right now it is only populating a single entry in the dictionary `table_create_child_attribute_map`. I'm assuming in the future, large query breakdown will have its corresponding entries in the dictionary but I don't see a way of populating using a single plan as input

Reply:
i think the question is probably not how do you propagate this map, but first do you need to propagate this map? If yes, then we can not skip schema query build, otherwise. Otherwise, we can always redesign the save_as_table interface to make sure things are optional

Reply:
as the offline discussion, the property only needs to be propagated if the table is created on top of file reading and when use insert statement. For the usage of large query breakdown, the breakdown shouldn't happen right after file reading, and there should be no need to propogate this property, we can redesign the interface of save_as_table when dealing with the large query breakdown.

Reply:
If it is only the top level node that needs to save the attributes, then why not save a single list of attributes instead of a map which is only going to have a single entry?

Reply:
yes, updated to have one entry